### PR TITLE
add java.time.Duration support in akka-actor module for the javadsl

### PR DESCRIPTION
**Pull request by ralphlaude**
Add java.time.Duration support for javadsl by adding the setReceiveTimeout and cancelReceiveTimeout in the class Abstract.ActorContext

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
Refs #24646

**Comment by johanandren** [comment]
Needs MiMa excludes as it is adding methods to a public API. See the [CONTRIBUTING.md section on binary compatibility](https://github.com/akka/akka/blob/master/CONTRIBUTING.md#binary-compatibility) for details

**Comment by ralphlaude** [comment]
@johanandren: Do it need something else here, except putting something in the MiMa excludes?

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Comment by patriknw** [comment]
@ralphlaude The binary compatibility check failed:
```
[04-30 09:09:26.376] [error]  * method cancelReceiveTimeout()Unit in trait akka.actor.AbstractActor#ActorContext is present only in current version
[04-30 09:09:26.376] [error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.AbstractActor#ActorContext.cancelReceiveTimeout")
[04-30 09:09:26.376] [error]  * method setReceiveTimeout(java.time.Duration)Unit in trait akka.actor.AbstractActor#ActorContext is present only in current version
[04-30 09:09:26.376] [error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.AbstractActor#ActorContext.setReceiveTimeout")
```

That is a change that we can do because ActorContext is not intended to be extended.
Please add the following to `akka-actor/src/main/mima-filters/2.5.12.backwards.excludes`:
```
# #24646 java.time.Duration
ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.AbstractActor#ActorContext.cancelReceiveTimeout")
ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.AbstractActor#ActorContext.setReceiveTimeout")
```

**Review by johanandren — APPROVED** [review]
LGTM with one minor thing

**Comment by ralphlaude** [comment]
Done

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 1]
```diff
@@ -755,7 +755,7 @@ public void usePatternsAskPipe() {
         ActorRef actorC = getRef();
 
         //#ask-pipe
-        Timeout t = new Timeout(Duration.create(5, TimeUnit.SECONDS));
+        Timeout t = new Timeout(scala.concurrent.duration.Duration.create(5, TimeUnit.SECONDS));
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: `Timeout.create(java-duration)` or `new Timeout(2, TimeUnit.SECONDS)`

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 2]
```diff
@@ -628,7 +628,7 @@ public void using_hot_swap() {
         actor.tell("foo", getRef());
         actor.tell("foo", getRef());
         expectMsgEquals("I am already angry?");
-        expectNoMsg(Duration.create(1, TimeUnit.SECONDS));
+        expectNoMsg(scala.concurrent.duration.Duration.create(1, TimeUnit.SECONDS));
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: There is a `expectNoMessage(java-duration)` in the Java testkit, use that.

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public static void beforeClass() {
 
   @AfterClass
   public static void afterClass() throws Exception {
-    Await.ready(system.terminate(), Duration.create(5, TimeUnit.SECONDS));
+    Await.ready(system.terminate(), scala.concurrent.duration.Duration.create(5, TimeUnit.SECONDS));
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: `TestKit.shutdownActorSystem(system);`

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 4]
```diff
@@ -791,7 +791,7 @@ public void useKill() {
         victim.tell(akka.actor.Kill.getInstance(), ActorRef.noSender());
         
         // expecting the actor to indeed terminate:
-        expectTerminated(Duration.create(3, TimeUnit.SECONDS), victim);
+        expectTerminated(scala.concurrent.duration.Duration.create(3, TimeUnit.SECONDS), victim);
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: This one we should add to the Java testkit with Java duration as well, but we can do that separately from this PR

**Review by johanandren — COMMENTED** [review]
Just a few more small things.

**Comment by ralphlaude** [comment]
@johanandren thanks for the hint. I will do it that way and i understand the context better.

**Comment by ralphlaude** [comment]
Done

File: akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala [hunk 5]
```diff
@@ -633,6 +633,17 @@ class TestKit(system: ActorSystem) {
...
+  /**
+   * Receive one message from the test actor and assert that it is the Terminated message of the given ActorRef.
+   * Before calling this method, you have to `watch` the target actor ref.
+   * Wait time is bounded by the given duration, with an AssertionFailure being thrown in case of timeout.
+   *
+   * @param max wait no more than max time, otherwise throw AssertionFailure
+   * @param target the actor ref expected to be Terminated
+   * @return the received Terminated message
+   */
+  def expectTerminated(max: java.time.Duration, target: ActorRef): Terminated = tp.expectTerminated(target, max.asScala)
```
Comment (johanandren) [inline akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala]: Ah, thanks for adding directly in this PR!

File: akka-docs/src/test/java/jdocs/actor/FaultHandlingDocSample.java [hunk 6]
```diff
@@ -67,7 +67,7 @@ public static void main(String[] args) {
     public void preStart() {
       // If we don't get any progress within 15 seconds then the service
       // is unavailable
-      getContext().setReceiveTimeout(Duration.create("15 seconds"));
+      getContext().setReceiveTimeout(java.time.Duration.ofSeconds(15));
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/actor/FaultHandlingDocSample.java]: Same as the other one, make this the default import and use a fqn for the Scala one if it is used

**Review by johanandren — COMMENTED** [review]
Just a tiny bit more feedback!

File: akka-docs/src/test/java/jdocs/actor/FaultHandlingDocSample.java (hunk 6 above)
Comment (ralphlaude) [reply akka-docs/src/test/java/jdocs/actor/FaultHandlingDocSample.java]: Sorry, i didn't see that the last time.

**Comment by ralphlaude** [comment]
I hope it is done this time

**Review by ktoso — APPROVED** [review]
LGTM, thanks

**Pull request merged**